elessar/pages/AddUserPage/AddUserPage.tsx

```diff
@@ -0,0 +1,348 @@
+import React, { Fragment, Component} from 'react';
+import { View, Keyboard, ScrollView, ActivityIndicator, StyleSheet, TouchableOpacity, Text, TextInput } from 'react-native';
+import { validateEmail, validatePassword, validatePhone, validateName, validateRole} from '../Login/validation';
+import { storeData } from '../Login/store';
+import { getToken, addUser } from '../Login/requests';
+import * as yup from 'yup';
+import { Formik } from 'formik'
+import moment from "moment";
+import DatePicker from '@react-native-community/datetimepicker';
+import {Picker} from '@react-native-community/picker';
+import { Input } from 'react-native-elements';
+
+
+
+interface AddUserPageState {
+    name: string,
+    phone: string,
+    birth: Date,
+    role: string,
+    email: string,
+    message: string,
+    password: string,
+    passwordMessage: string,
+    emailMessage: string,
+    phoneMessage: string,
+    nameMessage: string,
+    roleMessage: string,
+    loading: boolean,
+    showDatePicker: boolean,
+    dateMessage: string,
+    showRolePicker: boolean,
+}
+
+interface Props {
+    navigation: any
+  }
+
+export default class AddUserPage extends Component<Props, AddUserPageState>{
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: '',
+            phone:'',
+            birth: new Date(),
+            role: '',
+            email: '',
+            message: '',
+            passwordMessage: '',
+            emailMessage: '',
+            phoneMessage: '',
+            nameMessage: '',
+            roleMessage: '',
+            dateMessage: '',
+            password:'',
+            loading: false,
+            showDatePicker: false,
+            showRolePicker: false,
+        }
+    }
+
+    private handleButton = async () => {
+        const isEmailValid = validateEmail(this.state.email);
+        const isPasswordValid = validatePassword(this.state.password);
+        const isPhoneValid = validatePhone(this.state.phone);
+        const isNameValid = validateName(this.state.name);
+        const isRoleValid = validateRole(this.state.role);
+        
+        this.setState({loading: true})
```

Comment:
Acho que você poderia trocar o estado pra loading só quando for fazer a chamada. As outras operações são praticamente instantâneas à percepção humana.